### ISSUE TITLE
Update all check_auth calls to use into_val on tuples

### ISSUE THREAD

**Issue by sisuresh** [stellar/soroban-examples#80]
Ex.
```
check_auth(
            ...
            vec![&e, nonce.into_val(&e), num.clone().into_val(&e)],
        );
```
should be 
```
check_auth(
            ...
            (nonce, num.clone()).into_val(&e),
        );
```

**Comment by sisuresh** [comment]
This change requires an up to date SDK.

**Comment by sisuresh** [comment]
Closed by #85
